
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">

<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    <title>Walkthrough &#8212; gatk-pipelines 1 documentation</title>
    <link rel="stylesheet" href="_static/alabaster.css" type="text/css" />
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    <script type="text/javascript">
      var DOCUMENTATION_OPTIONS = {
        URL_ROOT:    './',
        VERSION:     '1',
        COLLAPSE_INDEX: false,
        FILE_SUFFIX: '.html',
        HAS_SOURCE:  true,
        SOURCELINK_SUFFIX: '.txt'
      };
    </script>
    <script type="text/javascript" src="_static/jquery.js"></script>
    <script type="text/javascript" src="_static/underscore.js"></script>
    <script type="text/javascript" src="_static/doctools.js"></script>
    <script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="prev" title="Documentation for Brown Core GATK Pipeline Deployments" href="index.html" />
   
  <link rel="stylesheet" href="_static/custom.css" type="text/css" />
  
  
  <meta name="viewport" content="width=device-width, initial-scale=0.9, maximum-scale=0.9" />

  </head>
  <body>
  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          <div class="body" role="main">
            
  <div class="section" id="walkthrough">
<h1>Walkthrough<a class="headerlink" href="#walkthrough" title="Permalink to this headline">¶</a></h1>
</div>
<div class="section" id="required-reference-files">
<h1>Required Reference Files<a class="headerlink" href="#required-reference-files" title="Permalink to this headline">¶</a></h1>
<p>The Broad Best Practices protocol incorporates a number of reference files, including things such as a reference genome and known snp locations.  These files can be found as a part of the ‘bundle’ associated with the genomic reference of the user’s choice.  For the purposes of this example, we elected to utilize hg19.</p>
<p>Here, we assign the various references - most of which have very long file paths due to the directory structure of the Broad bundle - to variables so that we can refer to them in a more readable fashion later:</p>
<div class="highlight-default"><div class="highlight"><pre><span></span><span class="n">myfasta</span><span class="o">=</span><span class="s2">&quot;/users/aleith/scratch/broad/ftp.broadinstitute.org/bundle/hg19/ucsc.hg19.fasta&quot;</span>
<span class="n">mydict</span><span class="o">=</span><span class="s2">&quot;/users/aleith/scratch/broad/ftp.broadinstitute.org/bundle/hg19/ucsc.hg19.fasta.dict&quot;</span>
<span class="n">mysnp</span><span class="o">=</span><span class="s2">&quot;/users/aleith/scratch/broad/ftp.broadinstitute.org/bundle/hg19/dbsnp_138.hg19.vcf&quot;</span>
<span class="n">myindel</span><span class="o">=</span><span class="s2">&quot;/users/aleith/scratch/broad/ftp.broadinstitute.org/bundle/hg19/Mills_and_1000G_gold_standard.indels.hg19.sites.vcf&quot;</span>
<span class="n">myhapmap</span><span class="o">=</span><span class="s2">&quot;/users/aleith/scratch/broad/ftp.broadinstitute.org/bundle/hg19/hapmap_3.3.hg19.sites.vcf&quot;</span>
<span class="n">myomni</span><span class="o">=</span><span class="s2">&quot;/users/aleith/scratch/broad/ftp.broadinstitute.org/bundle/hg19/1000G_omni2.5.hg19.sites.vcf&quot;</span>
<span class="n">mymills</span><span class="o">=</span><span class="s2">&quot;/users/aleith/scratch/broad/ftp.broadinstitute.org/bundle/hg19/Mills_and_1000G_gold_standard.indels.hg19.sites.vcf&quot;</span>
<span class="n">myhighconf</span><span class="o">=</span><span class="s2">&quot;/users/aleith/scratch/broad/ftp.broadinstitute.org/bundle/hg19/1000G_phase1.snps.high_confidence.hg19.sites.vcf&quot;</span>
</pre></div>
</div>
</div>
<div class="section" id="sample-processing-and-preparation">
<h1>Sample Processing and Preparation<a class="headerlink" href="#sample-processing-and-preparation" title="Permalink to this headline">¶</a></h1>
<p>To begin with, the raw reads must be aligned to a reference genome, ultimately generating a bam file for analysis by GATK.  The aligner preferred by the GATK developers is BWA, so that is the only aligner our protocol presently accounts for.</p>
<p>The following steps should be applied to each pair of fastq files; our protocol accomplishes this process by looping over every sample, though it should be possible to modify for parallel computation given sufficient resources.</p>
<p>We define the variable:</p>
<div class="highlight-default"><div class="highlight"><pre><span></span>$mysamplebase
myR1=$mysamplebase&quot;_1.fastq&quot;
myR2=$mysamplebase&quot;_2.fastq&quot;
</pre></div>
</div>
<p>as the base filename (e.g. ‘SRR062634’ for ‘SRR062634_1.fastq’ and ‘SRR062634_2.fastq’), and then use that to determine the names of the fastqs for the given sample.  This variable is used in nearly every successive function in the entire pipeline, so assigning it correctly is imperative.</p>
<p>Align reads to the reference genome using <a class="reference external" href="http://bio-bwa.sourceforge.net/bwa.shtml">BWA-mem</a>, specifying read groups as is mandatory:</p>
<div class="highlight-default"><div class="highlight"><pre><span></span>bwa mem \
  -t 8 \
  -R &#39;@RG\tID:&#39;$mysamplebase&#39;\tLB:&#39;$mysamplebase&#39;\tSM:&#39;$mysamplebase&#39;\tPL:ILLUMINA&#39; \
  $myfasta \
  $myR1 $myR2 \
  &gt; $mysamplebase&quot;.sam&quot;
</pre></div>
</div>
<p>Using <a class="reference external" href="http://www.htslib.org/doc/samtools.html">samtools</a>, process the SAM file into a sorted BAM file:</p>
<div class="highlight-default"><div class="highlight"><pre><span></span>samtools view -S -b $mysamplebase&quot;.sam&quot; &gt; $mysamplebase&quot;.bam&quot;
samtools sort $mysamplebase&quot;.bam&quot; -T temp -o $mysamplebase&quot;_sorted.bam&quot;
samtools index $mysamplebase&quot;_sorted.bam&quot;
</pre></div>
</div>
<p>Run Picard Tools’ <a class="reference external" href="https://broadinstitute.github.io/picard/command-line-overview.html#MeanQualityByCycle">MeanQualityByCycle</a> utility:</p>
<div class="highlight-default"><div class="highlight"><pre><span></span>picard MeanQualityByCycle \
  INPUT=$mysamplebase&quot;_sorted.bam&quot; \
  CHART_OUTPUT=$mysamplebase&quot;_mean_quality_by_cycle.pdf&quot; \
  OUTPUT=$mysamplebase&quot;_read_quality_by_cycle.txt&quot; \
  VALIDATION_STRINGENCY=LENIENT \
  REFERENCE_SEQUENCE=$myfasta
</pre></div>
</div>
<p>the <a class="reference external" href="https://broadinstitute.github.io/picard/command-line-overview.html#QualityScoreDistribution">QualityScoreDistribution</a> utility:</p>
<div class="highlight-default"><div class="highlight"><pre><span></span>picard QualityScoreDistribution \
  INPUT=$mysamplebase&quot;_sorted.bam&quot; \
  CHART_OUTPUT=$mysamplebase&quot;_mean_quality_overall.pdf&quot; \
  OUTPUT=$mysamplebase&quot;_read_quality_overall.txt&quot; \
  VALIDATION_STRINGENCY=LENIENT \
  REFERENCE_SEQUENCE=$myfasta
</pre></div>
</div>
<p>the <a class="reference external" href="https://broadinstitute.github.io/picard/command-line-overview.html#CollectWgsMetrics">CollectWgsMetrics</a> utility:</p>
<div class="highlight-default"><div class="highlight"><pre><span></span>picard CollectWgsMetrics \
  INPUT=$mysamplebase&quot;_sorted.bam&quot; OUTPUT=$mysamplebase&quot;_stats_picard.txt&quot; \
  REFERENCE_SEQUENCE=$myfasta \
  MINIMUM_MAPPING_QUALITY=20 \
  MINIMUM_BASE_QUALITY=20 \
  VALIDATION_STRINGENCY=LENIENT
</pre></div>
</div>
<p>and finally the <a class="reference external" href="https://broadinstitute.github.io/picard/command-line-overview.html#MarkDuplicates">MarkDuplicates</a> utility:</p>
<div class="highlight-default"><div class="highlight"><pre><span></span>picard MarkDuplicates I=$mysamplebase&quot;_sorted.bam&quot; \
  O=$mysamplebase&quot;_sorted_dedup.bam&quot; M=$mysamplebase&quot;_dedup_metric.txt&quot; \
  VALIDATION_STRINGENCY=LENIENT \
  REMOVE_DUPLICATES=true CREATE_INDEX=true
</pre></div>
</div>
<p>This last step, which produces a deduplicated BAM file, is particularly important for further analysis.  In each case, VALIDATION_STRINGENCY=LENIENT is passed to Picard Tools as recommended by the GATK developers on their forums; omission of this argument often leads to the software breaking.</p>
<p>Next, the first two GATK steps proceed, creating realignment targets and then utilizing them.</p>
<p>GATK’s <a class="reference external" href="https://software.broadinstitute.org/gatk/documentation/tooldocs/current/org_broadinstitute_gatk_tools_walkers_indels_RealignerTargetCreator.php">RealignerTargetCreator</a>:</p>
<div class="highlight-default"><div class="highlight"><pre><span></span>gatk -T RealignerTargetCreator \
  -R $myfasta \
  -I $mysamplebase&quot;_sorted_dedup.bam&quot; \
  -known $myindel \
  -o $mysamplebase&quot;_realign_targets.intervals&quot;

gatk -T IndelRealigner \
  -R $myfasta \
  -known $myindel \
  -targetIntervals $mysamplebase&quot;_realign_targets.intervals&quot;  \
  -I $mysamplebase&quot;_sorted_dedup.bam&quot; \
  -o $mysamplebase&quot;_sorted_dedup_realigned.bam&quot; \
  --filter_bases_not_stored
</pre></div>
</div>
<p>The resulting BAM file is then indexed:</p>
<div class="highlight-default"><div class="highlight"><pre><span></span>picard BuildBamIndex \
  I=$mysamplebase&quot;_sorted_dedup_realigned.bam&quot;
</pre></div>
</div>
<p>GATK <a class="reference external" href="https://software.broadinstitute.org/gatk/documentation/tooldocs/current/org_broadinstitute_gatk_tools_walkers_bqsr_BaseRecalibrator.php">BaseRecalibrator</a>:</p>
<div class="highlight-default"><div class="highlight"><pre><span></span>gatk -T BaseRecalibrator \
  -R $myfasta \
  -knownSites $mysnp \
  -knownSites $myindel \
  -I $mysamplebase&quot;_sorted_dedup_realigned.bam&quot; \
  -o $mysamplebase&quot;_recal_table.txt&quot; \
  -nct 8

gatk -T BaseRecalibrator \
  -R $myfasta \
  -I $mysamplebase&quot;_sorted_dedup_realigned.bam&quot; \
  -knownSites $mysnp \
  -knownSites $myindel \
  -BQSR $mysamplebase&quot;_recal_table.txt&quot; \
  -o $mysamplebase&quot;_post_recal_table.txt&quot;
</pre></div>
</div>
<p>Picard <a class="reference external" href="https://broadinstitute.github.io/picard/command-line-overview.html#CollectOxoGMetrics">CollectOxoGMetrics</a>:</p>
<div class="highlight-default"><div class="highlight"><pre><span></span>picard CollectOxoGMetrics \
  I=$mysamplebase&quot;_sorted_dedup_realigned.bam&quot; \
  O=$mysamplebase&quot;_sorted_dedup_realigned_oxoG_metrics.txt&quot; \
  R=$myfasta \
  VALIDATION_STRINGENCY=LENIENT
</pre></div>
</div>
<p>Picard <a class="reference external" href="https://broadinstitute.github.io/picard/command-line-overview.html#CollectSequencingArtifactMetrics">CollectSequencingArtifactMetrics</a>:</p>
<div class="highlight-default"><div class="highlight"><pre><span></span>picard CollectSequencingArtifactMetrics \
  I=$mysamplebase&quot;_sorted_dedup_realigned.bam&quot; \
  O=$mysamplebase&quot;_sorted_dedup_realigned_artifact_metrics.txt&quot; \
  R=$myfasta \
  VALIDATION_STRINGENCY=LENIENT
</pre></div>
</div>
<p>Picard <a class="reference external" href="https://broadinstitute.github.io/picard/command-line-overview.html#CollectAlignmentSummaryMetrics">CollectAlignmentSummaryMetrics</a>:</p>
<div class="highlight-default"><div class="highlight"><pre><span></span>picard CollectAlignmentSummaryMetrics \
  R=$myfasta \
  I=$mysamplebase&quot;_sorted_dedup_realigned.bam&quot; \
  O=$mysamplebase&quot;_sorted_dedup_realigned_alignment_metrics.txt&quot;
</pre></div>
</div>
<p>Picard <a class="reference external" href="https://broadinstitute.github.io/picard/command-line-overview.html#CollectInsertSizeMetrics">CollectInsertSizeMetrics</a>:</p>
<div class="highlight-default"><div class="highlight"><pre><span></span>picard CollectInsertSizeMetrics \
  I=$mysamplebase&quot;_sorted_dedup_realigned.bam&quot; \
  O=$mysamplebase&quot;_sorted_dedup_realigned_insert_size_metrics.txt&quot; \
  H=$mysamplebase&quot;_sorted_dedup_realigned_insert_size_histogram.pdf&quot; \
  M=0.5
</pre></div>
</div>
<p>GATK <a class="reference external" href="https://software.broadinstitute.org/gatk/gatkdocs/3.5-0/org_broadinstitute_gatk_tools_walkers_bqsr_AnalyzeCovariates.php">AnalyzeCovariates</a>:</p>
<div class="highlight-default"><div class="highlight"><pre><span></span>gatk -T AnalyzeCovariates \
  -R $myfasta \
  -before $mysamplebase&quot;_recal_table.txt&quot; \
  -after $mysamplebase&quot;_post_recal_table.txt&quot; \
  -plots $mysamplebase&quot;_recalibration_plots.pdf&quot;
</pre></div>
</div>
<p>GATK <a class="reference external" href="https://software.broadinstitute.org/gatk/documentation/tooldocs/current/org_broadinstitute_gatk_tools_walkers_readutils_PrintReads.php">PrintReads</a>:</p>
<div class="highlight-default"><div class="highlight"><pre><span></span>gatk -T PrintReads \
  -R $myfasta \
  -I $mysamplebase&quot;_sorted_dedup_realigned.bam&quot; \
  -BQSR $mysamplebase&quot;_recal_table.txt&quot; \
  -o  $mysamplebase&quot;_GATK.bam&quot;
</pre></div>
</div>
<p>Picard <a class="reference external" href="https://broadinstitute.github.io/picard/command-line-overview.html#BuildBamIndex">BuildBamIndex</a>:</p>
<div class="highlight-default"><div class="highlight"><pre><span></span>picard BuildBamIndex \
  I= $mysamplebase&quot;_GATK.bam&quot;
</pre></div>
</div>
<p>GATK <a class="reference external" href="https://software.broadinstitute.org/gatk/documentation/tooldocs/current/org_broadinstitute_gatk_tools_walkers_haplotypecaller_HaplotypeCaller.php">HaplotypeCaller</a>:</p>
<div class="highlight-default"><div class="highlight"><pre><span></span>gatk -Xmx64g -T HaplotypeCaller --disable_auto_index_creation_and_locking_when_reading_rods \
  -R $myfasta \
  -I  $mysamplebase&quot;_GATK.bam&quot; \
  --dbsnp  $mysnp \
  -stand_call_conf 30 \
  --minPruning 3 --maxNumHaplotypesInPopulation 200 \
  --max_alternate_alleles 3 -variant_index_parameter 128000 \
  -variant_index_type LINEAR -contamination 0.0 \
  --genotyping_mode DISCOVERY \
  --emitRefConfidence GVCF \
  -o $mysamplebase&quot;_GATK-HC.g.vcf&quot;
</pre></div>
</div>
</div>
<div class="section" id="combined-sample-analysis">
<h1>Combined-Sample Analysis<a class="headerlink" href="#combined-sample-analysis" title="Permalink to this headline">¶</a></h1>
<p>Generate a list of samples:</p>
<div class="highlight-default"><div class="highlight"><pre><span></span><span class="n">ls</span> <span class="o">-</span><span class="mi">1</span><span class="n">a</span> <span class="o">*</span><span class="n">GATK</span><span class="o">-</span><span class="n">HC_exac</span><span class="o">.</span><span class="n">g</span><span class="o">.</span><span class="n">vcf</span> <span class="o">&gt;</span> <span class="n">vcfs_exac</span><span class="o">.</span><span class="n">list</span>
</pre></div>
</div>
<p>GATK <a class="reference external" href="https://software.broadinstitute.org/gatk/documentation/tooldocs/current/org_broadinstitute_gatk_tools_walkers_variantutils_GenotypeGVCFs.php">GenotypeGVCFs</a>:</p>
<div class="highlight-default"><div class="highlight"><pre><span></span><span class="n">gatk</span> <span class="o">-</span><span class="n">T</span> <span class="n">GenotypeGVCFs</span> \
  <span class="o">-</span><span class="n">R</span> <span class="s2">&quot;/users/aleith/scratch/broad/ftp.broadinstitute.org/bundle/hg19/ucsc.hg19.fasta&quot;</span> \
  <span class="o">--</span><span class="n">variant</span> <span class="s2">&quot;vcfs_exac.list&quot;</span> \
  <span class="o">-</span><span class="n">o</span> <span class="s2">&quot;all_GATK-HC_joined_exac.g.vcf&quot;</span>
</pre></div>
</div>
<p>Picard <a class="reference external" href="https://broadinstitute.github.io/picard/command-line-overview.html#MakeSitesOnlyVcf">MakeSitesOnlyVcf</a>:</p>
<div class="highlight-default"><div class="highlight"><pre><span></span><span class="n">picard</span> <span class="n">MakeSitesOnlyVcf</span> \
  <span class="n">INPUT</span><span class="o">=</span><span class="s2">&quot;all_GATK-HC_joined_exac.g.vcf&quot;</span> \
  <span class="n">OUTPUT</span><span class="o">=</span><span class="s2">&quot;all_GATK-HC_joined_sites_only_exac.g.vcf&quot;</span> \
  <span class="n">VALIDATION_STRINGENCY</span><span class="o">=</span><span class="n">LENIENT</span>
</pre></div>
</div>
<p>GATK <a class="reference external" href="https://software.broadinstitute.org/gatk/documentation/tooldocs/current/org_broadinstitute_gatk_tools_walkers_variantrecalibration_VariantRecalibrator.php">VariantRecalibrator</a>:</p>
<div class="highlight-default"><div class="highlight"><pre><span></span>gatk -T VariantRecalibrator \
  -R &quot;/users/aleith/scratch/broad/ftp.broadinstitute.org/bundle/hg19/ucsc.hg19.fasta&quot; \
  -input &quot;all_GATK-HC_joined_sites_only_exac.g.vcf&quot; \
  -resource:hapmap,known=false,training=true,truth=true,prior=15.0 /users/aleith/scratch/broad/ftp.broadinstitute.org/bundle/hg19/hapmap_3.3.hg19.sites.vcf \
  -resource:omni,known=false,training=true,truth=true,prior=12.0 /users/aleith/scratch/broad/ftp.broadinstitute.org/bundle/hg19/1000G_omni2.5.hg19.sites.vcf \
  -resource:1000G,known=false,training=true,truth=false,prior=10.0 /users/aleith/scratch/broad/ftp.broadinstitute.org/bundle/hg19/1000G_phase1.snps.high_confidence.hg19.sites.vcf \
  -resource:dbsnp,known=true,training=false,truth=false,prior=2.0 $mysnp \
  --maxGaussians 6 \
  -an QD -an MQRankSum -an ReadPosRankSum -an FS -an MQ -an InbreedingCoeff \
  -mode SNP \
  -allPoly -tranche 100.0 -tranche 99.95 -tranche 99.9 -tranche 99.8 -tranche 99.6 \
  -tranche 99.5 -tranche 99.4 -tranche 99.3 -tranche 99.0 -tranche 98.0 -tranche 97.0 \
  -tranche 90.0 \
  -recalFile &quot;all_recalibrate_SNP_exac.recal&quot; \
  -tranchesFile &quot;all_recalibrate_SNP_exac.tranches&quot; \
  -rscriptFile &quot;all_recalibrate_SNP_plots_exac.R&quot;
</pre></div>
</div>
<p>GATK <a class="reference external" href="https://software.broadinstitute.org/gatk/documentation/tooldocs/current/org_broadinstitute_gatk_tools_walkers_variantrecalibration_ApplyRecalibration.php">ApplyRecalibration</a>:</p>
<div class="highlight-default"><div class="highlight"><pre><span></span><span class="n">gatk</span> <span class="o">-</span><span class="n">T</span> <span class="n">ApplyRecalibration</span> \
  <span class="o">-</span><span class="n">R</span> <span class="s2">&quot;/users/aleith/scratch/broad/ftp.broadinstitute.org/bundle/hg19/ucsc.hg19.fasta&quot;</span> \
  <span class="o">-</span><span class="nb">input</span> <span class="s2">&quot;all_GATK-HC_joined_sites_only_exac.g.vcf&quot;</span> \
  <span class="o">-</span><span class="n">mode</span> <span class="n">SNP</span> \
  <span class="o">--</span><span class="n">ts_filter_level</span> <span class="mf">99.6</span> \
  <span class="o">-</span><span class="n">recalFile</span> <span class="s2">&quot;all_recalibrate_SNP_exac.recal&quot;</span> \
  <span class="o">-</span><span class="n">tranchesFile</span> <span class="s2">&quot;all_recalibrate_SNP_exac.tranches&quot;</span> \
  <span class="o">-</span><span class="n">o</span> <span class="s2">&quot;all_recalibrated_snps_raw_indels_exac.vcf&quot;</span>
</pre></div>
</div>
<p>GATK <a class="reference external" href="https://software.broadinstitute.org/gatk/documentation/tooldocs/current/org_broadinstitute_gatk_tools_walkers_variantrecalibration_VariantRecalibrator.php">VariantRecalibrator</a>:</p>
<div class="highlight-default"><div class="highlight"><pre><span></span>gatk -T VariantRecalibrator \
  -R &quot;/users/aleith/scratch/broad/ftp.broadinstitute.org/bundle/hg19/ucsc.hg19.fasta&quot; \
  -input &quot;all_recalibrated_snps_raw_indels_exac.vcf&quot; \
  -resource:mills,known=false,training=true,truth=true,prior=12.0 /users/aleith/scratch/broad/ftp.broadinstitute.org/bundle/hg19/Mills_and_1000G_gold_standard.indels.hg19.sites.vcf \
  -resource:dbsnp,known=true,training=false,truth=false,prior=2.0 $mysnp \
  -an FS -an ReadPosRankSum -an InbreedingCoeff -an MQRankSum -an QD \
  -mode INDEL \
  -allPoly -tranche 100.0 -tranche 99.95 -tranche 99.9 -tranche 99.5 -tranche 99.0 -tranche 97.0 \
  -tranche 96.0 -tranche 95.0 -tranche 94.0 -tranche 93.5 -tranche 93.0 -tranche 92.0 -tranche 91.0 -tranche 90.0 \
  --maxGaussians 6 \
  -recalFile &quot;all_recalibrate_INDEL_exac.recal&quot; \
  -tranchesFile &quot;all_recalibrate_INDEL_exac.tranches&quot; \
  -rscriptFile &quot;all_recalibrate_INDEL_plots_exac.R&quot;
</pre></div>
</div>
<p>GATK <a class="reference external" href="https://software.broadinstitute.org/gatk/documentation/tooldocs/current/org_broadinstitute_gatk_tools_walkers_variantrecalibration_ApplyRecalibration.php">ApplyRecalibration</a>:</p>
<div class="highlight-default"><div class="highlight"><pre><span></span><span class="n">gatk</span> <span class="o">-</span><span class="n">T</span> <span class="n">ApplyRecalibration</span> \
  <span class="o">-</span><span class="n">R</span> <span class="s2">&quot;/users/aleith/scratch/broad/ftp.broadinstitute.org/bundle/hg19/ucsc.hg19.fasta&quot;</span> \
  <span class="o">-</span><span class="nb">input</span> <span class="s2">&quot;all_recalibrated_snps_raw_indels_exac.vcf&quot;</span> \
  <span class="o">-</span><span class="n">mode</span> <span class="n">INDEL</span> \
  <span class="o">--</span><span class="n">ts_filter_level</span> <span class="mf">95.0</span> \
  <span class="o">-</span><span class="n">recalFile</span> <span class="s2">&quot;all_recalibrate_INDEL_exac.recal&quot;</span> \
  <span class="o">-</span><span class="n">tranchesFile</span> <span class="s2">&quot;all_recalibrate_INDEL_exac.tranches&quot;</span> \
  <span class="o">-</span><span class="n">o</span> <span class="s2">&quot;all_recalibrated_variants_exac.vcf&quot;</span>
</pre></div>
</div>
</div>


          </div>
        </div>
      </div>
      <div class="sphinxsidebar" role="navigation" aria-label="main navigation">
        <div class="sphinxsidebarwrapper">
<h1 class="logo"><a href="index.html">gatk-pipelines</a></h1>








<h3>Navigation</h3>
<p class="caption"><span class="caption-text">Contents:</span></p>
<ul class="current">
<li class="toctree-l1 current"><a class="current reference internal" href="#">Walkthrough</a></li>
<li class="toctree-l1"><a class="reference internal" href="#required-reference-files">Required Reference Files</a></li>
<li class="toctree-l1"><a class="reference internal" href="#sample-processing-and-preparation">Sample Processing and Preparation</a></li>
<li class="toctree-l1"><a class="reference internal" href="#combined-sample-analysis">Combined-Sample Analysis</a></li>
</ul>

<div class="relations">
<h3>Related Topics</h3>
<ul>
  <li><a href="index.html">Documentation overview</a><ul>
      <li>Previous: <a href="index.html" title="previous chapter">Documentation for Brown Core GATK Pipeline Deployments</a></li>
  </ul></li>
</ul>
</div>
<div id="searchbox" style="display: none" role="search">
  <h3>Quick search</h3>
    <form class="search" action="search.html" method="get">
      <div><input type="text" name="q" /></div>
      <div><input type="submit" value="Go" /></div>
      <input type="hidden" name="check_keywords" value="yes" />
      <input type="hidden" name="area" value="default" />
    </form>
</div>
<script type="text/javascript">$('#searchbox').show(0);</script>
        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="footer">
      &copy;2017, Andrew Leith.
      
      |
      Powered by <a href="http://sphinx-doc.org/">Sphinx 1.6.3</a>
      &amp; <a href="https://github.com/bitprophet/alabaster">Alabaster 0.7.10</a>
      
      |
      <a href="_sources/gatk.rst.txt"
          rel="nofollow">Page source</a>
    </div>

    

    
  </body>
</html>